### DISCUSSION TITLE
Castle screen, Marketpace window, Select Recipients window, position of "OKAY" and "CANCEL" buttons

### DISCUSSION THREAD

**Discussion by LeHerosInconnu** [ihhub/fheroes2#4623] [Ideas]
I suggest to change the position of these buttons so that they are not "stuck" in the corners of the frame.
A placement similar to their position in the creature recruitment window in relation to the frame would be nice.

In fheroes2:

![Marketplace gift 002](https://user-images.githubusercontent.com/43583994/141466539-ee8f785f-3c99-4a3f-966c-718f420a1bb5.png)

![Marketplace gift 004](https://user-images.githubusercontent.com/43583994/141467011-b6fb89e1-afac-48e8-b75d-1844c0f833d8.png)

**Comment by a1exsh** [comment]
I think this will be fixed by #4350

**Reply by zenseii** [reply]
Was just about to ask you about this. Does the PR fix the issue where if you press down any of these two buttons they leave an imprint on the background of this dialog window?

Here's an image for reference if you look at the top right and left corners:
![image](https://user-images.githubusercontent.com/12501091/141471136-5f336432-b2be-46cc-8a32-668c08fb63da.png)

**Reply by zenseii** [reply]
Just read through your PR and saw that you're already working on this 👍

**Reply by LeHerosInconnu** [reply]
@a1exsh.

> I think this will be fixed by #4350

Wow, you did a great job with all those buttons. :)

**Comment by LeHerosInconnu** [comment]
There has been an improvement, but the buttons are still too close to the edges.

In fheroes2 (1.1.10 9995):

<img width="656" height="518" alt="Marketplace gift window buttons position 01" src="https://github.com/user-attachments/assets/74f5e62f-70d5-4627-8772-3dde3d8c4528" />

**Comment by LeHerosInconnu** [comment]
This is planned in my mod: https://github.com/ihhub/fheroes2/discussions/10132.
I close the discussion.
